### PR TITLE
Modified lines-from-chunks generator to ensure support for "\r\n"

### PR DESCRIPTION
**Pull request by nlw0**
The current code produces empty strings sometimes if the input file follows this unconventional and bizarre "\r\n" newline standard, that nonetheless is still found in some legacy systems. The proposed code tries to make sure whatever `splitlines()` understands as newline is supported.

This PR is the same as #1527, but rebasing after lots of changes upstream.

### CODE REVIEW

File: tests/unit/test_response.py [hunk 1]
```diff
@@ -158,6 +158,15 @@ def stream(self, amt=1024, decode_content=None):
                 break
             yield chunk
 
+    def test_streaming_line_abstruse_newline_standard(self):
```
Comment (jamesls) [inline tests/unit/test_response.py]: It looks like the rebase got messed up here.  This test is now part of the `FakeRawResponse` class so it won't be run.  I think this was meant to be part of `TestStreamWrapper`.

Comment (nlw0) [reply tests/unit/test_response.py]: Oops, sorry for the messy rebase. Just made a new patch

**Comment by nlw0** [comment]
@jamesls @joguSD  I moved back the test. Is it OK now?

**Comment by nlw0** [comment]
Still waiting for news, maybe it's because I didn't "resolve the conversation"?... Sorry!

**Comment by nlw0** [comment]
So, is there anything else I might do to land this?

**Comment by jamesls** [comment]
Sorry for the delay, looking at this now.

**Pull request merged**